### PR TITLE
Bookstore: use Redux in React components.

### PR DESCRIPTION
**Pull request by MarcoOquendoC**
# [Bookstore: use Redux in React components.](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/use_redux_in_react_components.md)

## Hi good reviewer,

P and Bob thinking

### Project requirements

- Keep state immutable in reducers.
- Import Redux Provider.
- Connect the app to the Redux store.
- Use useSelector to consume the state.
- Add functionality for "Add new".
- Add a functionality for "Remove".
- Add a functionality for "Check status"

### CODE REVIEW

File: src/redux/categories/categories.js [hunk 1]
```diff
@@ -4,12 +4,16 @@ const CHECK_STATUS = 'CHECK_STATUS';
...
+const categoryReducer = (state = initialState, action) => {
   switch (action.type) {
     case CHECK_STATUS:
       return [...state, action.payload];
     default: return state;
   }
-}
+};
+
+export default categoryReducer;
```
Comment (cwaku) [inline src/redux/categories/categories.js:19]: - [ ] Good job creating a reducer to check the status but you are missing one more thing. In your React component responsible for categories, add a functionality for "Check status" button click:
  - Import Redux `useDispatch`.
  - Dispatch a corresponding action (import Action Creator that you created before and use it here to display the state on the page).

**Review by cwaku — CHANGES_REQUESTED** [review]
Hi @MarcoOquendoC 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!
There are a few issues 🐛  that you still need to work 🛠️  on to go to the next project ⏭️, but you are almost there! 🏃🏾 

## Highlights
 - [x] Correct GitHub Flow 👍🏾 
 - [x] Good commit history 🥇 
 - [x] Good PR title/description 💯 
 - [x] All linters passed 🥇 

## Required Changes ♻️
It's ok to have bugs in your code. Please, see this code review as a chance to learn practical coding skills and new tips 🌟  that can improve your code quality 💯 and help you get a better job 💼.

I will guide 📔 you through fixing 🛠️ these bugs, follow my comments down below and make sure you don't miss any of them to get your project approved ✅

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categories.js (hunk 1 above)
Comment (MarcoOquendoC) [reply src/redux/categories/categories.js:19]: Hi Ricky, @cwaku 

Thanks, the functionality has been added.

**Review by mikethreels — APPROVED** [review]
Hi Marco, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job following the instructions from the previous CR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If You have submitted a PR which pulls from the main into Dev this is incorrect. it should be the other way around you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**